webgpu.h

```diff
@@ -881,6 +894,7 @@ typedef WGPUProc (*WGPUProcGetProcAddress)(WGPUDevice device, char const * procN
 
 // Procs of Adapter
 typedef void (*WGPUProcAdapterGetProperties)(WGPUAdapter adapter, WGPUAdapterProperties * properties);
+typedef uint32_t (*WGPUProcAdapterGetFeatures)(WGPUAdapter adapter, WGPUFeatureName * features);
```

Comment:
Are the features retrieved as a separate entrypoint or do we augment `WGPUAdapterProperties`? I kind of had the idea that `WGPUAdapterProperties` would be a structure that contains everything, and could also be augmented with a chain, so that the application would look into the chain similarly to what the implementation does.

Reply:
Wouldn't we need to still have a separate entrypoint to find out how many features there are so the application can allocate enough memory to hold the list? So it would sort of be the same thing but populating the list inside a chained struct. 

Overall I like it separate to match the JS API better.

Reply:
The `WGPUAdapterProperties` could have a `featureNameCount` and a `featureNames` members. The pointers would have the same lifetime as the `WGPUAdapter`. But I don't have a super strong opinion at the moment.

Reply:
Ah I see. I do see that we probably don't need to have the two-call mechanism to get the size and populate. The adapter probably needs to store the list anyway.

I still think it should be a separate entrypoint though to match JS better.

```suggestion
typedef void (*WGPUProcAdapterGetFeatures)(WGPUAdapter adapter, WGPUFeatureName ** features, uint32_t * featuresCount);
```

Reply:
Sounds good, I'm just we're going to look at the adapter API again in the future when we get closer to a 1.0 webgpu.h

Reply:
Writing down `WGPUFeatureName**` seems to be quite weird. I don't think that's close to what any of the native APIs do.

Reply:
Well, we could keep the two-call mechanism then. Either with the previous suggestion or like 

```c
typedef void (*WGPUProcAdapterGetFeatures)(WGPUAdapter adapter, WGPUFeatureName ** features, uint32_t * featuresCount);
```

this is what [vkEnumerateDeviceExtensionProperties](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkEnumerateDeviceExtensionProperties.html) does.

Reply:
Another option is to just add `featureCount` field to `WGPUAdapterProperties`.

Reply:
Yes, and Corentin even suggested putting *both* the pointer and the count on `WGPUAdapterProperties`.  I don't really feel that strongly about this, but the count is analogous to asking for `adapter.features.length` in JS and not `adapter.properties.featuresLength`, so it makes more sense to me to keep the pointer to the features list and the size of the features list in the same place.

Reply:
Maybe this would a bit divergent, but this seems like a very reasonable place to use a null-terminated list. Do we have other places in the C API that output lists?

Reply:
I would not expect the other places, no

Reply:
I forgot about this for a long time. I changed this so that support for features is queried individually. I realized it's not as useful to have a list when we'd want a more set-like interface. This better matches GPUSupportedFeatures in the JS API, and my suggested native limit API in #102.

Reply:
The WebIDL setlike thing is still iterable though.